modules/yazi/module.nix

```diff
@@ -0,0 +1,65 @@
+{
+  wlib,
+  lib,
+  ...
+}:
+wlib.wrapModule (
+  { config, ... }:
+  let
+    tomlFmt = config.pkgs.formats.toml { };
+  in
+  {
+    options = {
+      settings = lib.mkOption {
+        type = tomlFmt.type;
+        default = { };
+        description = ''
+          Yazi configuration for yazi.toml.
+          See https://yazi-rs.github.io/docs/configuration/yazi
+        '';
+      };
+
+      keymap = lib.mkOption {
+        type = tomlFmt.type;
+        default = { };
+        description = ''
+          Yazi configuration for keymap.toml.
+          See <https://yazi-rs.github.io/docs/configuration/keymap>
+        '';
+      };
+
+      theme = lib.mkOption {
+        type = tomlFmt.type;
+        default = { };
+        description = ''
+          Yazi configuration for theme.toml.
+          See <https://yazi-rs.github.io/docs/configuration/theme>
+        '';
+      };
+
+      extraFlags = lib.mkOption {
+        type = lib.types.attrsOf lib.types.unspecified; # TODO add list handling
+        default = { };
+        description = "Extra flags to pass to yazi.";
+      };
+    };
+
+    config =
+      let
+        settingsToml = tomlFmt.generate "yazi.toml" config.settings;
+        keymapToml = tomlFmt.generate "keymap.toml" config.keymap;
+        themeToml = tomlFmt.generate "theme.toml" config.theme;
```

Comment:
would be nice if those files could also be exported via a wlib.types.file option. Take a look here https://github.com/Lassulus/wrappers/blob/main/modules/mpv/module.nix#L15